project/Build.scala

```diff
@@ -322,8 +322,12 @@ object Build {
       }
   )
 
-  val noClassFilesSettings: Setting[_] =
-    scalacOptions in (Compile, compile) += "-Yskip:cleanup,icode,jvm"
+  val noClassFilesSettings: Setting[_] = {
+    scalacOptions in (Compile, compile) += {
+      if (isGeneratingForIDE) "-Yskip:jvm"
```

Comment:
Why do we need to tell the IDE what phases to skip?

Reply:
Metals uses a phase after `jvm`, so we can't just `-Ystop-after`, or that phase will also be skipped, and some important features of Metals are lost (in particular, go-to-definition). We have to skip `jvm` because it crashes in the `javalanglib`.